### PR TITLE
[GCU] Support gcu platform

### PR DESCRIPTION
**Pull request by EnflameGCU**
新增`GCU`硬件适配，包括：

- 编译工程
- `platforms`
- `gcu ops`
- `gcu_model_runner/gcu_worker`
- `attention`
- `moe`
- `quantization`
- `activation/normalization/linear/rotary_embedding/sampler`
- 其他流程适配等

目前支持的模型：

- ERNIE-4.5-300B-A47B
- ERNIE-4.5-21B-A3B
- ERNIE-4.5-0.3B

详细安装使用请参见[使用说明](https://github.com/PaddlePaddle/FastDeploy/blob/develop/docs/get_started/installation/Enflame_gcu.md)。

### CODE REVIEW

File: fastdeploy/model_executor/layers/backends/gcu/attention/mem_efficient_attn_backend.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                value=v_.unsqueeze(0),
+                attn_mask=None,
+                dropout=0.0,
+                softmax_scale=self.scaling,
+                mask_mode=1,
+                seqlens=[0],
+                causal=self.causal,
+            )
+
+            # res = self.native_sdpa_impl(
```
Comment (vivienfanghuagood) [inline fastdeploy/model_executor/layers/backends/gcu/attention/mem_efficient_attn_backend.py]: remove these useless codes

**Review by vivienfanghuagood — COMMENTED** [review]
Please add some detaied information in PR:
- supported model lists & tutorials
- user guide about build & install GCU Fastdeploy

File: fastdeploy/model_executor/layers/backends/gcu/attention/mem_efficient_attn_backend.py (hunk 1 above)
Comment (EnflameGCU) [reply fastdeploy/model_executor/layers/backends/gcu/attention/mem_efficient_attn_backend.py]: mem_efficient_attn_backend其中的native sdpa实现主要用来调测中精度对比，期望的是保留该部分代码，调测新网络时方便本地快速切换，已切换到通过变量控制

**Comment by EnflameGCU** [comment]
> Please add some detaied information in PR:
> 
> * supported model lists & tutorials
> * user guide about build & install GCU Fastdeploy

已提供，详见PR第一条comment

File: fastdeploy/worker/gcu_model_runner.py [hunk 2]
```diff
@@ -0,0 +1,1232 @@
...
+        elif request.structural_tag is not None:
+            schemata_key = ("structural_tag", request.structural_tag)
+
+        return self.guided_backend.get_logits_processor(
+            schemata_key=schemata_key), schemata_key
+
+    def insert_prefill_inputs(self, req_dicts: List[Request]):
+        """
+        Process inputs for prefill tasks and insert it to share_inputs buffer
+        TODO(gongshaotian): Refactor this func
```
Comment (yongqiangma) [inline fastdeploy/worker/gcu_model_runner.py]: 多个内容的注释是gongshaotian，根据实际情况修改下吧

File: fastdeploy/model_executor/layers/normalization.py [hunk 3]
```diff
@@ -128,19 +137,32 @@ def forward(
...
-            quant_max_bound=self.quant_max_bound,
-            quant_min_bound=self.quant_min_bound,
-        )
+        if current_platform.is_gcu():
+            if residual_input is None:
+                return rms_norm(x, self.ln_weight, self.eps)
+            norm_out = self.norm_func(
+                x, residual_input, self.ln_weight, self.eps
+            )
+            # if residual_input is not None:
```
Comment (yongqiangma) [inline fastdeploy/model_executor/layers/normalization.py]: 这些注释主要删除。

File: fastdeploy/worker/gcu_model_runner.py [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+            max_num_blocks=max_block_num)
+        # local_rank = self.local_rank % self.parallel_config.tensor_parallel_degree
+
+        if not self.parallel_config.do_profile and (
+                self.parallel_config.enable_prefix_caching \
+                or self.parallel_config.splitwise_role != "mixed"):
+            raise NotImplementedError(
+                "prefix_caching is not support by GCUModelRunner."
+            )
+            # cache_kvs_list = []
```
Comment (yongqiangma) [inline fastdeploy/worker/gcu_model_runner.py]: 注释掉的内容需要删除

File: fastdeploy/worker/gcu_model_runner.py (hunk 2 above)
Comment (EnflameGCU) [reply fastdeploy/worker/gcu_model_runner.py]: 已修改

File: fastdeploy/model_executor/layers/normalization.py (hunk 3 above)
Comment (EnflameGCU) [reply fastdeploy/model_executor/layers/normalization.py]: 已删除

File: fastdeploy/worker/gcu_model_runner.py (hunk 4 above)
Comment (EnflameGCU) [reply fastdeploy/worker/gcu_model_runner.py]: 已删除

**Review by yongqiangma — APPROVED** [review]
